### PR TITLE
Expand webserver timeline

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Webserver now reads timelines from xml and parses them with ejs

### CODE REVIEW

File: src/main/webserver/routes/timeline.js [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
 router.get("/", (req, res) => {
-    // Dummy array here
-    // Todo: replace with real things,
-    // note: the array must be ordered or be sorted here
-    const timelineBlocks = [];
-    timelineBlocks.push(new TimelineBlock(3, 5));
-    timelineBlocks.push(new TimelineBlock(7, 12));
-    timelineBlocks.push(new TimelineBlock(15, 22));
+    // Dummyfile Todo: replace with dynamic
+    fs.readFile(`${__dirname}/../test_project.scp`, (err, data) => {
```
Comment (geenen124) [inline src/main/webserver/routes/timeline.js:12]: Pretty big method... Consider abstracting to other methods.

File: src/main/webserver/views/timeline.ejs [hunk 2]
```diff
@@ -21,32 +20,68 @@
...
+                            <% }); %>
                         </tr>
                         <%
-                            var curBlockNumber = 0;
-                            var prevBlock = null;
-                            var curBlock = timelineBlocks[curBlockNumber];
-                            for(i = minCount; i <= maxCount; i++) {
-                                var nextBlock = timelineBlocks %>
+                            // Setup vars
+                            var curShotNumber = new Array();
```
Comment (martijnjanssen) [inline src/main/webserver/views/timeline.ejs]: Why not use `[]`? This looks quite sketchy.

File: src/main/webserver/views/timeline.ejs [hunk 3]
```diff
@@ -21,32 +20,68 @@
...
+                                            } else {
+                                                curShotNumber[j] = -1;
+                                                prevShot[j] = curShot[j];
+                                                curShot[j] = null;
+                                            }
+
+                                            // Skip td if we are in a block, otherwise insert.
+                                            // DONT TOUCH THESE TESTS, THEY BREAK VERY EASILY
+                                        } else if (prevShot[j] === null || (curShot[j] !== null && curShot[j].beginCount*4 > i
+                                                    && prevShot[j].endCount*4 < i )
```
Comment (martijnjanssen) [inline src/main/webserver/views/timeline.ejs]: No whitespace around operators? I would think that `*` is more readable.

**Comment by martijnjanssen** [comment]
Looks good!

**Pull request merged**